accepted/2024/support-nuget-authentication-plugins-dotnet-tools.md

```diff
@@ -0,0 +1,210 @@
+
+Another motivating factor for this work is the current story for installing these credential providers, which is not ideal. For instance, let's consider the two cross-platform authentication plugins that I am aware of while writing this specification:
+
+1. **Azure Artifacts Credential Provider** - The [setup](https://github.com/microsoft/artifacts-credprovider/tree/master?tab=readme-ov-file#setup) instructions vary based on the platform.
+2. **AWS CodeArtifact Credential Provider** - The AWS team has developed a [.NET tool](https://docs.aws.amazon.com/codeartifact/latest/ug/nuget-cli.html#nuget-configure-cli) to facilitate authentication with their private NuGet feeds. In their current implementation, they've added a subcommand, `codeartifact-creds install`, which copies the credential provider to the NuGet plugins folder. The log below shows all possible subcommands. However, plugin authors could leverage the .NET SDK to allow their customers to install, uninstall, or update the NuGet plugins more efficiently.
+
+```log
+~\.nuget\plugins\tools
+❯ .\dotnet-codeartifact-creds.exe
+Required command was not provided.
+
+Usage:
+  dotnet-codeartifact-creds [options] [command]
+
+Options:
+  --version         Show version information
+  -?, -h, --help    Show help and usage information
+
+Commands:
+  install      Installs the AWS CodeArtifact NuGet credential provider into the NuGet plugins folder.
+  configure    Sets or Unsets a configuration
+  uninstall    Uninstalls the AWS CodeArtifact NuGet credential provider from the NuGet plugins folder.
+```
+
+## Explanation
+
+### Functional explanation
+
+A deployment solution for .NET is the [.NET tools](https://learn.microsoft.com/dotnet/core/tools). These tools provide a seamless installation and management experience for NuGet packages in the .NET ecosystem. The use of .NET tools as a deployment mechanism has been a recurring request from users and internal partners who need to authenticate with private repositories. Currently, this solution works for the Windows .NET Framework. However, the goal is to extend this support cross-platform for all supported .NET runtimes.
+
+By implementing this specification, we offer plugin authors the option to use .NET tools for plugin deployment. They can install these as a `tool path` global tool. This eliminates the need to maintain separate versions for `.NET Framework` and `.NET Core`. It also simplifies the installation process by removing the necessity for plugin authors to create subcommands like `codeartifact-creds install/uninstall`.
+
+The ideal workflow for repositories that access private NuGet feeds, such as Azure DevOps, would be:
```

Comment:
+1 on this, from the customer (and us supporting users trying to set this up) this is less than ideal. This seems to be only ideal from the implementation side as tools are installed in a NuGet specific directory so there's conflict with other tools.

Reply:
> This seems to be only ideal from the implementation side as tools are installed in a NuGet specific directory so there's conflict with other tools.

Hi John, thank you for the feedback. I assume you meant there is **no** conflict with other tools because they tools are installed to a NuGet specific directory. Is that correct?

Reply:
yup, type there, updated!